hibernate-core/src/test/java/org/hibernate/test/type/LocalDateTimeTest.java

```diff
@@ -0,0 +1,209 @@
+			LocalDateTime.parse( "1900-01-01T00:19:32" ).atZone( ZoneId.of( "Europe/Amsterdam" ) ), // HHH-13266 breaking point
+			LocalDateTime.parse( "1900-01-01T01:00:00" ).atZone( ZoneId.of( "Europe/Amsterdam" ) ),
+			LocalDateTime.parse( "1899-12-31T23:39:03" ).atZone( ZoneId.of( "Europe/Amsterdam" ) )
+	);
+
+	private static Dialect determineDialect() {
+		try {
+			return Dialect.getDialect();
+		}
+		catch (Exception e) {
+			return new Dialect() {
+			};
+		}
+	}
+
+	@Parameterized.Parameters(name = "{0}")
+	public static List<Object[]> data() {
+		Dialect dialect = determineDialect();
+		List<Object[]> results = new ArrayList<>();
+		for ( ZonedDateTime zonedDateTime : ZONED_DATE_TIMES ) {
+			results.add( new Object[] { zonedDateTime } );
+			/*
+			 * The bug reported in HHH-13266 seems to appear before a certain date/time for each time zone (called "breaking point").
+			 * Some of the dates we're looping on represent the first date/time that works fine for a certain time zone.
+			 * Let's also try some dates before, so that we trigger the bug.
+			 */
+			results.add( new Object[] { zonedDateTime.minus( 10L, ChronoUnit.HOURS ) } );
+			results.add( new Object[] { zonedDateTime.minus( 1L, ChronoUnit.SECONDS ) } );
+			if ( !( dialect instanceof PostgreSQL81Dialect ) ) {
+				// PostgreSQL apparently doesn't support nanosecond precision correctly
+				results.add( new Object[] { zonedDateTime.minus( 1L, ChronoUnit.NANOS ) } );
+			}
+		}
+		return results;
+	}
+
+	private final ZoneId zoneId;
+	private final LocalDateTime localDateTime;
+
+	public LocalDateTimeTest(ZonedDateTime zonedDateTime) {
+		this.zoneId = zonedDateTime.getZone();
+		this.localDateTime = zonedDateTime.toLocalDateTime();
+	}
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[] { EntityWithLocalDateTime.class };
+	}
+
+	@Before
+	public void cleanup() {
+		inTransaction( session -> {
+			session.createNativeQuery(
+					"DELETE FROM theentity"
+			)
+					.executeUpdate();
+		} );
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-13266")
+	public void writeThenRead() {
+		withDefaultTimeZone( zoneId, () -> {
+			inTransaction( session -> {
+				session.persist( new EntityWithLocalDateTime( 1, localDateTime ) );
+			} );
+			inTransaction( session -> {
+				LocalDateTime read = session.find( EntityWithLocalDateTime.class, 1 ).value;
+				assertEquals(
+						"Writing then reading a value should return the original value",
+						localDateTime, read
+				);
+			} );
+		} );
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-13266")
+	public void writeThenNativeRead() {
+		withDefaultTimeZone( zoneId, () -> {
+			inTransaction( session -> {
+				session.persist( new EntityWithLocalDateTime( 1, localDateTime ) );
+			} );
+			inTransaction( session -> {
+				Timestamp nativeRead = (Timestamp) session.createNativeQuery(
+						"SELECT thevalue FROM theentity WHERE theid = :id"
+				)
+						.setParameter( "id", 1 )
+						.uniqueResult();
```

Comment:
I just want to mention that, in general, when Hibernate executes a native query, the return type may not be exactly what gets returned by raw JDBC. In this particular case, it is.

Hibernate uses ```ResultSetMetaData``` to determine the SQL type of the column. In this case, the column is of type ```java.sql.Types.TIMESTAMP```. By default, ```java.sql.Types.TIMESTAMP``` is mapped to ```TimestampType.INSTANCE```, which delegates to ```JdbcTimestampTypeDescriptor``` for wrapping and unwrapping. When a value of type ```java.sql.Timestamp``` is wrapped or unwrapped, the value is simply cast to ```java.sql.Timestamp```.

I needed to walk through this in the debugger to convince myself that Hibernate was not converting the value that got returned by the database in any way.

This test appears to safely do what it needs to do, but I would prefer to see the native query executed and the return value obtained using raw JDBC in an implementation of the ```org.hibernate.jdbc.Work``` API.